streams/shared/src/main/scala/zio/stream/experimental/ZSink.scala

```diff
@@ -253,6 +253,111 @@ abstract class ZSink[-R, +E, -I, +Z] private (
       }
     } yield push)
 
+  /**
+   * Runs both sinks in parallel on the input and combines the results
+   * using the provided function.
+   */
+  final def zipWithPar[R1 <: R, E1 >: E, I1 <: I, Z1, Z2](
+    that: ZSink[R1, E1, I1, Z1]
+  )(f: (Z, Z1) => Z2): ZSink[R1, E1, I1, Z2] = {
+
+    type State = (Option[Z], Option[Z1])
+
+    def applyLeft(s: State, res: Either[Either[E, Z], Unit]): ZIO[R1, Either[E1, Z2], State] =
```

Comment:
You can call the `applyLeft/Right` functions only when `Either[E, Z/Z2]` is received to save the outer either.